tikv/mvcc.go

```diff
@@ -108,15 +125,40 @@ func (store *MVCCStore) Close() error {
 	return nil
 }
 
-func (store *MVCCStore) getLatestTS() uint64 {
-	return atomic.LoadUint64(&store.latestTS)
+type lockEntryHdr struct {
+	keyLen uint32
+	valLen uint32
 }
 
-func (store *MVCCStore) updateLatestTS(ts uint64) {
-	latestTS := store.getLatestTS()
-	if ts != math.MaxUint64 && ts > latestTS {
-		atomic.CompareAndSwapUint64(&store.latestTS, latestTS, ts)
+func (store *MVCCStore) dumpMemLocks() error {
+	tmpFileName := store.dir + "/lock_store.tmp"
+	f, err := os.OpenFile(tmpFileName, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0666)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	writer := bufio.NewWriter(f)
+	cnt := 0
+	it := store.lockStore.NewIterator()
+	hdrBuf := make([]byte, 8)
+	hdr := (*lockEntryHdr)(unsafe.Pointer(&hdrBuf[0]))
+	for it.SeekToFirst(); it.Valid(); it.Next() {
+		hdr.keyLen = uint32(len(it.Key()))
+		hdr.valLen = uint32(len(it.Value()))
+		writer.Write(hdrBuf)
+		writer.Write(it.Key())
+		writer.Write(it.Value())
+		cnt++
```

Comment:
Seems cnt not used?

Reply:
I didn't modify the code, just move around, maybe later we can use it to debug.